src/components/app/MenuButtons.js

```diff
@@ -534,63 +535,87 @@ class ProfileSharingCompositeButton extends React.PureComponent<
           currentButtonIsSecondaryShareButton: isSecondaryShareButtonVisible,
         })}
       >
-        <ProfileSharingButton
-          buttonClassName="menuButtonsShareButton"
-          shareLabel={shareLabel}
-          symbolicationStatus={symbolicationStatus}
-          okButtonClickEvent={this._attemptToShare}
-          shareNetworkUrlCheckboxChecked={this.state.shareNetworkUrls}
-          shareNetworkUrlCheckboxOnChange={this._onChangeShareNetworkUrls}
-          checkboxDisabled={false}
-        />
-        <UploadingStatus progress={uploadProgress} />
-        <ButtonWithPanel
-          className="menuButtonsPermalinkButton"
-          ref={this._takePermalinkButtonRef}
-          label="Permalink"
-          panel={
-            <ArrowPanel
-              className="menuButtonsPermalinkPanel"
-              onOpen={this._onPermalinkPanelOpen}
-              onClose={this._onPermalinkPanelClose}
-            >
-              <input
-                type="text"
-                className="menuButtonsPermalinkTextField"
-                value={shortUrl}
-                readOnly="readOnly"
-                ref={this._takePermalinkTextFieldRef}
+        {/* display only the button that is visible */}
+        {state === 'local' ? (
+          <CSSTransition classNames="animateDown" timeout={200} in={true}>
+            <ProfileSharingButton
+              buttonClassName="menuButtonsShareButton"
+              shareLabel={shareLabel}
+              symbolicationStatus={symbolicationStatus}
+              okButtonClickEvent={this._attemptToShare}
+              shareNetworkUrlCheckboxChecked={this.state.shareNetworkUrls}
+              shareNetworkUrlCheckboxOnChange={this._onChangeShareNetworkUrls}
+              checkboxDisabled={false}
+            />
+          </CSSTransition>
+        ) : null}
+
+        {state === 'uploading' || state === 'public' ? (
+          <CSSTransition classNames="animateDown" timeout={0} in={true}>
```

Comment:
If we keep this, I think we can extract this `<CSSTransition timeout={0} ` component into a separate component (it doesn't need to be in another file, it can be in the same file); for example name it `<NoTransition>` :). Also, is the `classnames` needed?